### PR TITLE
add task for adding repository keys via file

### PR DESCRIPTION
**Pull request by muellerbe**
Signed-off-by: Bernd Mueller <mueller@b1-systems.de>

### CODE REVIEW

File: roles/repository/defaults/main.yml [hunk 1]
```diff
@@ -1 +1,3 @@
 ---
+repository_keys_file: []
+repository_keys_file_dir: "/opt/configuration/environments/generic/repokeys"
```
Comment (berendt) [inline roles/repository/defaults/main.yml:3]: Is this parameter required? It is not used in the ``Add repository keys via file`` task.

File: roles/repository/defaults/main.yml [hunk 2]
```diff
@@ -1 +1,3 @@
 ---
+repository_keys_file: []
```
Comment (berendt) [inline roles/repository/defaults/main.yml:2]: ``repository_key_files`` makes more sense to me as parameter name.

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/repository/defaults/main.yml (hunk 1 above)
Comment (muellerbe) [reply roles/repository/defaults/main.yml]: thanks for the hint, I want to use this, see in task in the new commit

File: roles/repository/defaults/main.yml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 ---
-repository_keys_file: []
-repository_keys_file_dir: "/opt/configuration/environments/generic/repokeys"
+repository_key_files: []
+repository_key_files_dir: "/opt/configuration/environments/generic/repokeys"
```
Comment (berendt) [inline roles/repository/defaults/main.yml:3]: I would prefer to have only one parameter. Either _files with full path or only _dir and then all files in _dir are treated as keys and imported.

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/repository/defaults/main.yml (hunk 3 above)
Comment (muellerbe) [reply roles/repository/defaults/main.yml]: with_items -> with_fileglob

File: roles/repository/defaults/main.yml [hunk 4]
```diff
@@ -1,3 +1,2 @@
 ---
-repository_key_files: []
-repository_key_files_dir: "/opt/configuration/environments/generic/repokeys"
+repository_key_files_directory: "/opt/configuration/environments/generic/repokeys"
```
Comment (berendt) [inline roles/repository/defaults/main.yml:2]: As mentioned in my first review please use ``files/repository/keys`` here.

**Review by berendt — CHANGES_REQUESTED** [review]

**Comment by berendt** [comment]
Replaced by #294.

File: roles/repository/defaults/main.yml (hunk 4 above)
Comment (muellerbe) [reply roles/repository/defaults/main.yml:2]: where do you mention this? I cannot find.
do you mean /opt/configuration/environments/generic/files/repository/keys ?

**Pull request closed without merging**